models/receipt.go

```diff
@@ -0,0 +1,32 @@
+package models
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"math/big"
+)
+
+// StorageReceipt is a receipt representation for storage.
+//
+// This struct copies the geth.Receipt type found here: https://github.com/ethereum/go-ethereum/blob/9bbb9df18549d6f81c3d1f4fc6c65f71bc92490d/core/types/receipt.go#L52
+// the reason is if we use geth.Receipt some values will be skipped when RLP encoding which is because
+// geth node has the data locally, but we don't in evm gateway, so we can not reproduce those values
+// and we need to store them
+type StorageReceipt struct {
+	Type              uint8
+	PostState         []byte
+	Status            uint64
+	CumulativeGasUsed uint64
+	// todo we could skip bloom to optimize storage and dynamically recalculate it
```

Comment:
Good point. That is easy to do with:
```go
types.LogsBloom(logs)
```
I am not sure if there is a performance overhead to re-calculate it for each endpoint that has to return it.